update_handlers.go

```diff
@@ -0,0 +1,131 @@
+        return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"message": "User not found"})
+    }
+
+    user.Username = username
+    if err := database.DB.Save(&user).Error; err != nil {
+        return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"message": "Couldn't update username"})
+    }
+
+    return c.JSON(fiber.Map{"message": "Username updated successfully"})
+}
+
+// UpdateEmail updates a user's email
+func UpdateEmail(c *fiber.Ctx) error {
+    var data map[string]string
+
+    if err := c.BodyParser(&data); err != nil {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Invalid input"})
+    }
+
+    userID := data["user_id"]
+    email := data["email"]
+    confirmEmail := data["confirm_email"]
+
+    if email != confirmEmail {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Emails do not match"})
+    }
+
+    var user entity.Account
+    if err := database.DB.First(&user, userID).Error; err != nil {
+        return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"message": "User not found"})
+    }
+
+    user.Email = email
+    if err := database.DB.Save(&user).Error; err != nil {
+        return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"message": "Couldn't update email"})
+    }
+
+    return c.JSON(fiber.Map{"message": "Email updated successfully"})
+}
+
+// RevokeToken invalidates the token by setting its expiration time to the past
+func RevokeToken(tokenString string) error {
+    secretKey := []byte(os.Getenv("JWT_SECRET"))
+
+    token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+        return secretKey, nil
+    })
+
+    if err != nil || !token.Valid {
+        return err
+    }
+
+    claims := token.Claims.(jwt.MapClaims)
+    claims["exp"] = time.Now().Add(-time.Hour).Unix() // Expire the token an hour ago
+
+    return nil
+}
+
+// UpdatePassword updates a user's password and logs them out
+func UpdatePassword(c *fiber.Ctx) error {
+    var data map[string]string
+
+    if err := c.BodyParser(&data); err != nil {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Invalid input"})
+    }
+
+    userID := data["user_id"]
+    password := data["password"]
+    confirmPassword := data["confirm_password"]
+    tokenString := c.Get("Authorization")
+
+    if password != confirmPassword {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Passwords do not match"})
+    }
+
+    var user entity.Account
+    if err := database.DB.First(&user, userID).Error; err != nil {
+        return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"message": "User not found"})
+    }
+
+    hashedPassword, err := utils.HashPassword(password)
+    if err != nil {
+        return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"message": "Couldn't hash password"})
+    }
+
+    user.Password = hashedPassword
```

Comment:
The problem here, this will not work because of how you handle the hashPassword above. A way to fix this, is to check if the user exists in the database (based on user ID), if it it does, then assign the new plain text password to the user, then call ``HashPassword(&user)``. Check if the password matches before assigning the new password to the user.Password.

Example:
```go
	var user entity.Account
	if err := database.DB.First(&user, userID).Error; err != nil {
		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"message": "User not found"})
	}

	if password != confirmPassword {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Passwords do not match"})
	}

	user.Password = password

	if err := utils.HashPassword(&user); err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"message": "Couldn't hash password"})
	}
```